tests/ut/test_core_setup.py

```diff
@@ -0,0 +1,265 @@
+        self._test_init_application_server_success(SlaveAuthenticaionMethod.PrivateKey)
+
+    def _test_init_application_server_success(self, authentication_method):
+        self.patch_call("time.sleep")
+        self._init_global_admin()
+        get_function = mock.MagicMock(side_effect=[
+            TestCoreSetup._generate_status_response(SetupWizardStage.Server, SetupWizardStatus.NA, ''),
+            TestCoreSetup._generate_status_response(SetupWizardStage.Replication, SetupWizardStatus.NA, ''),
+            TestCoreSetup._generate_status_response(SetupWizardStage.Finish, SetupWizardStatus.NA, '')
+        ])
+        execute_function = mock.MagicMock(side_effect=TestCoreSetup._create_init_slave_execute_function(authentication_method))
+        self._global_admin.get = get_function
+        self._global_admin.execute = execute_function
+        mock_startup_wait = self.patch_call("cterasdk.core.startup.Startup.wait")
+
+        setup.Setup(self._global_admin).init_application_server(self._master_ipaddr, self._master_secret)
+
+        setup_status_url = '/%s/setup/status' % (self._global_admin.context)
+        self._global_admin.get.assert_has_calls(
+            [
+                mock.call(setup_status_url, use_file_url=True),
+                mock.call(setup_status_url, use_file_url=True),
+                mock.call(setup_status_url, use_file_url=True)
+            ]
+        )
+        self._global_admin.execute.assert_has_calls(
+            [
+                mock.call('/%s/setup/authenticaionMethod' % (self._global_admin.context),
+                          'askMasterForSlaveAuthenticaionMethod', self._master_ipaddr, use_file_url=True),
+                mock.call('/%s/public/servers' % (self._global_admin.context), 'setReplication', mock.ANY, use_file_url=True)
+            ]
+        )
+        expected_param = TestCoreSetup._get_init_replication_param()
+        actual_param = self._global_admin.execute.call_args_list[1][0][2]  # Access setReplication call param
+        self._assert_equal_objects(actual_param, expected_param)
+
+        params = TestCoreSetup._get_init_server_params(ServerMode.Slave, authentication_method, self._master_ipaddr, self._master_secret)
+        form_data = TestCoreSetup._get_form_data(params)
+        self._global_admin.multipart.assert_called_once_with('/%s/setup' % (self._global_admin.context), form_data, use_file_url=True)
+        mock_startup_wait.assert_called_once()
+
+    def test_init_replication_server_success_password(self):
+        self._test_init_replication_server_success(SlaveAuthenticaionMethod.Password)
+
+    def test_init_replication_server_success_pk(self):
+        self._test_init_replication_server_success(SlaveAuthenticaionMethod.PrivateKey)
+
+    def _test_init_replication_server_success(self, authentication_method):
+        self.patch_call("time.sleep")
+        self._init_global_admin()
+        get_function = mock.MagicMock(side_effect=[
```

Comment:
Same, no need for these variables

Reply:
Done. Did that to overcome Pylint's 140 character limit